model/DataObject.php

```diff
@@ -1926,13 +1926,21 @@ public function many_many_extraFields($component = null) {
 				$manyMany = $SNG_class->stat('belongs_many_many');
 				$candidate = (isset($manyMany[$component])) ? $manyMany[$component] : null;
 				if($candidate) {
+					$relationName = null;
+					// Extract class and relation name from dot-notation
+					if(strpos($candidate, '.') !== false) {
+						list($candidate, $relationName) = explode('.', $candidate);
```

Comment:
You need to explicitly set the 3rd argument of explode to be `2` as you're using list.

```php
explode('.', $candidate, 2);
```